### PR TITLE
Grafana 2.0.2

### PR DESCRIPTION
**Pull request by mavimo**
This formula let build grafana 2.0.0-beta1 using brew.

How to use:
```bash
# Install grafana through brew
brew install grafana
# Create a sample config file
wget https://raw.githubusercontent.com/grafana/grafana/master/conf/sample.ini
# Start grafana web UI
grafana --config sample.ini web
# Open the UI on browser
open http://localhost:3000/
```

### CODE REVIEW

**Comment by torkelo** [comment]
The dependencies should not be in formula, `godep restore` will fetch all dependencies. 

This just builds the backend, not the frontend. Currently grafana binary needs to be run from a working directory relative to the public folder where the frontend assets are. And the frontend assets needs to be built, at least the less  -> css, but for optimized build the the complete `grunt build` or grunt `release step` are required.

**Comment by torkelo** [comment]
oh, and if you run go run build.go build  it will use dependencies from the repository itself. The error you mentioned on gitter was because grafana itself was not checkout out in the correct gopath (%GOPATH%/src/github.com/grafana/grafana)

**Comment by mikemcquaid** [comment]
> The dependencies should not be in formula, godep restore will fetch all dependencies.

That's not correct. They formula needs the dependencies because we don't want random revisions of all the dependencies fetched at `install` time but consistent revisions we've specified.

File: Library/Formula/grafana.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    ENV["GOHOME"] = buildpath
+
+    mkdir_p buildpath/"src/github.com/grafana/"
+    ln_sf buildpath, buildpath/"src/github.com/grafana/grafana"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "main.go"
+    bin.install "main" => "grafana"
+  end
+end
```
Comment (mikemcquaid) [inline Library/Formula/grafana.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/grafana.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
+require "language/go"
+
+class Grafana < Formula
+  homepage "http://www.grafana.org/"
+  head "https://github.com/grafana/grafana.git"
+  version "2.0.0-beta1"
+
+  stable do
+    url "https://github.com/grafana/grafana/archive/v2.0.0-beta1.tar.gz"
```
Comment (mikemcquaid) [inline Library/Formula/grafana.rb]: We need to use a stable e.g. non-beta version for the stable block.

**Comment by torkelo** [comment]
But the formula contains all grafana go dependencies and their revisions, those are already specified in the Godep json file in the grafana repository, and using go run build.go will use those dependencies, or godep restore will fetch exactly those dependencies & revisions specified in godep json file into the current go path. So either of those two options (using go run build.go or godep restore) is preferable, having all the go dependencies is in the formula seems to duplicate the exact same info already in specified in this file: https://github.com/grafana/grafana/blob/master/Godeps/Godeps.json

File: Library/Formula/grafana.rb (hunk 2 above)
Comment (torkelo) [inline Library/Formula/grafana.rb]: For Grafana 2.0 there currently is no stable release (latest is beta1). Grafana 2.0 is a substantial update compared to 1.x (new Go backend, etc). Can there be non stable homebrew packages?

Comment (mikemcquaid) [inline Library/Formula/grafana.rb]: Yes but they either need a stable block too or need to start in the `homebrew-devel-only` tap.

Comment (mavimo) [inline Library/Formula/grafana.rb]: Fixed c16701b9565b6ff44369dedad3b09e2358b4d4a6

**Comment by tdsmith** [comment]
We expect that information to be duplicated in the formula.

If this doesn't have a stable release, we shouldn't accept it to Homebrew/homebrew yet. It sounds like it's missing assets as well, unless it makes sense for those to be packaged separately?

**Comment by DomT4** [comment]
What's the status here?

**Comment by mavimo** [comment]
@DomT4 I'm going to rework it in the next week to use the stable release of grafana and introduce the frontend build.

**Comment by DomT4** [comment]
Thanks for the reply! That's fine. Ping us when you've updated the PR.

**Comment by jboursiquot** [comment]
:+1: Looking forward this this

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by mikemcquaid** [comment]
Please fix the audit warnings: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/29058/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_grafana___strict/

**Comment by DomT4** [comment]
Ping?

**Comment by mavimo** [comment]
@DomT4 I'm on vacation, I need to fix/improve tests and then can be merged. Current recipe work as expected on my machine.

**Comment by tpitale** [comment]
@mavimo Ping?

**Comment by tpitale** [comment]
I'm seeing this error when trying to build grafana with this recipe locally:

```
==> Cloning https://github.com/rainycape/unidecode.git
Cloning into '/Library/Caches/Homebrew/grafana--gopkgs.com-unidecode.v1--git'...
remote: Counting objects: 34, done.
remote: Total 34 (delta 0), reused 0 (delta 0), pack-reused 34
Unpacking objects: 100% (34/34), done.
Checking connectivity... done.
==> Checking out revision 4deae2c05236b41cc39f8144ac87a837ba974d40
==> npm install
==> npm install grunt-cli
==> ./node_modules/.bin/grunt
==> go build main.go

src/github.com/gosimple/slug/slug.go:13:2: cannot find package "github.com/rainycape/unidecode" in any of:
	/usr/local/Cellar/go/1.5/libexec/src/github.com/rainycape/unidecode (from $GOROOT)
	/private/tmp/grafana20150905-34475-leob0l/grafana-2.0.2/src/github.com/rainycape/unidecode (from $GOPATH)
src/github.com/macaron-contrib/session/redis/redis.go:26:2: code in directory /private/tmp/grafana20150905-34475-leob0l/grafana-2.0.2/src/gopkg.in/redis.v2 expects import "gopkg.in/redis.v3"

READ THIS: https://git.io/brew-troubleshooting

These open issues may also help:
Grafana 2.0.2 https://github.com/Homebrew/homebrew/pull/38247
```

**Comment by mavimo** [comment]
@tpitale confirmed, I'm working on fix it, thx for feedback!

**Comment by bfontaine** [comment]
Any update here?

**Comment by bfontaine** [comment]
Closing due to no response. Feel free to re-submit a working PR, thanks!

**Pull request closed without merging**